### PR TITLE
Enable full tests on all supported curves

### PR DESCRIPTION
**Pull request by dtebbs**
Enable running the full tests (currently run for pull requests only) on BLS12-377.

### CODE REVIEW

File: libzeth/tests/core/ec_operation_data_test.cpp [hunk 1]
```diff
@@ -15,6 +15,30 @@ using namespace libzeth;
...
+// coordinate.
+template<typename GroupT>
+static GroupT g1_curve_point_at_x(const typename GroupT::base_field &x)
+{
+    const typename GroupT::base_field x_squared = x * x;
+    const typename GroupT::base_field x_cubed = x_squared * x;
+    const typename GroupT::base_field y_squared =
+        x_cubed + (GroupT::coeff_a * x_squared) + GroupT::coeff_b;
+    const typename GroupT::base_field y = y_squared.sqrt();
+    return GroupT(x, y, GroupT::base_field::one());
```
Comment (AntoineRondelet) [inline libzeth/tests/core/ec_operation_data_test.cpp:28]: See: https://github.com/clearmatics/libff/pull/31/files#r543460055 - let's check that a root can be extracted (i.e. compute Legendre symbol) before calling `sqrt`

Comment (dtebbs) [reply libzeth/tests/core/ec_operation_data_test.cpp:28]: Thanks.  If the util functions are merged (see https://github.com/clearmatics/libff/pull/31#discussion_r544923921) we can change this to reference them instead.

Comment (dtebbs) [reply libzeth/tests/core/ec_operation_data_test.cpp]: Replaced by util fuctions in libff

**Review by AntoineRondelet — APPROVED** [review]
LGTM, thanks

**Comment by AntoineRondelet** [comment]
[Waiting before merging]
If the libsodium issue is now fixed (see comments in #332) #325 will need to remove the changes related to that. This PR will be merged after.

**Comment by dtebbs** [comment]
> [Waiting before merging]
> If the libsodium issue is now fixed (see comments in #332) #325 will need to remove the changes related to that. This PR will be merged after.

As mentioned in https://github.com/clearmatics/zeth/pull/332#issuecomment-747621907 I think this is worth including anyway.  This isn't the first time I've seen this particular problem, and relying on fewer servers in order to build can only be good.

**Pull request merged**